bundles/org.openhab.binding.luftdateninfo/src/main/java/org/openhab/binding/luftdateninfo/internal/handler/HTTPHandler.java

```diff
@@ -0,0 +1,182 @@
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.luftdateninfo.internal.handler;
+
+import java.util.Date;
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.openhab.binding.luftdateninfo.internal.dto.SensorData;
+import org.openhab.binding.luftdateninfo.internal.dto.SensorDataValue;
+import org.openhab.binding.luftdateninfo.internal.utils.DateTimeUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * The {@link HTTPHandler} is responsible for HTTP requests and JSON handling
+ *
+ * @author Bernd Weymann - Initial contribution
+ */
+@NonNullByDefault
+public class HTTPHandler {
+    private static final Logger logger = LoggerFactory.getLogger(HTTPHandler.class);
+    private static String sensorUrl = "http://data.sensor.community/airrohr/v1/sensor/";
+    private static @Nullable HttpClient commonHttpClient;
+
+    public static final String P1 = "P1";
+    public static final String P2 = "P2";
+
+    public static final String TEMPERATURE = "temperature";
+    public static final String HUMIDITY = "humidity";
+    public static final String PRESSURE = "pressure";
+    public static final String PRESSURE_SEALEVEL = "pressure_at_sealevel";
+
+    public static final String NOISE_EQ = "noise_LAeq";
+    public static final String NOISE_MIN = "noise_LA_min";
+    public static final String NOISE_MAX = "noise_LA_max";
+
+    public static void init(HttpClient httpClient) {
+        commonHttpClient = httpClient;
+    }
+
+    public static @Nullable String getResponse(String sensorId) {
+        if (commonHttpClient == null) {
+            logger.warn("HTTP Client not initialized");
+            return null;
+        }
+        String url = sensorUrl + sensorId + "/";
+        try {
+            ContentResponse contentResponse = commonHttpClient.newRequest(url).timeout(10, TimeUnit.SECONDS).send();
+            int httpStatus = contentResponse.getStatus();
+            String content = contentResponse.getContentAsString();
+            logger.debug("Sensor response: {}", httpStatus);
+            switch (httpStatus) {
+                case 200:
+                    return content;
+                case 400:
+                case 401:
+                case 404:
+                    logger.info("Sensor response: {}", httpStatus);
```

Comment:
Logging to info should be used rarely. This could be debug or warn.
Don't you want to log the other error codes, too? E.g. 500. You could log in the default case of the switch.

Reply:
Switched to WARN level